packages/hooks/src/useNumberFormatter.ts

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react'
+
+export const useNumberFormatter = ( initNumber : number, initLocale : string, initCurrency : string, initStyle : string) => {
```

Comment:
What if this was just a proxy for Intl.NumberFormat that returned a number formatter?

```
export function useNumberFormatter(locale?: string, options?: Intl.NumberFormatOptions): Intl.NumberFormat {
   const numberFormatter = React.useMemo( () => {
           return new Intl.NumberFormat(locale, options);
    }, [locale, options])

return numberFormatter
}
```
then it returns a `numberFormat` that could be used for formatting numbers?

There's just one challenge here in that the user needs to know to memoize `options` otherwise we'll get an infinite set of re-renders.

To work around that, we might just need to instantiate a new formatter for each format call:
```
export function useNumberFormatter(
  locale?: string,
  options?: Intl.NumberFormatOptions
) {
  const optionsRef = React.useRef(options);
  optionsRef.current = options;
  const numberFormatter = React.useMemo(() => {
    return function formatNumber(number: number) {
      const numberFormat = new Intl.NumberFormat(locale, optionsRef.current);
      return numberFormat.format(number);
    };
  }, [locale]);

  return numberFormatter;
}
```

now we're just wrapping Intl.NumberFormat here, but it should allow us in the future to add some defaults to handle `NaN` and `undefined` as well as provide some more default options when we want to format a currency.

Reply:
OK I'll try to edit it